core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -580,6 +605,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeLong(version);
             out.writeByte(state.id);
             Settings.writeSettingsToStream(settings, out);
+            out.writeIntArray(primaryTerms);
```

Comment:
Since we expect these to be non-negative and "not large", I wonder if it'd be better to serialize these using a variable-length encoding? See this [PR](https://github.com/elastic/elasticsearch/pull/14087).